ros2_batch_job/__main__.py

```diff
@@ -569,7 +569,7 @@ def run(args, build_function, blacklisted_package_names=None):
             job.run(
                 ['"%s"' % job.python, '-m', 'pip', 'uninstall', '-y'] +
                 ['cryptography', 'lxml', 'numpy'], shell=True)
-        pip_cmd = ['"%s"' % job.python, '-m', 'pip', 'install', '-U']
+        pip_cmd = ['"%s"' % job.python, '-m', 'pip', 'install', '-U', '--progress-bar off']
```

Comment:
While this does make it quieter, it has two issues:

1.  It doesn't work on Ubuntu 18.04.  While this isn't supported on Foxy, we still need to be able to run this for Eloquent and Dashing, so we need to make this conditional on the Ubuntu version.
1.  We don't want to *totally* disable the progress bar, as that doesn't give us any indication that pip hung on a download.  I might suggest switching this to `ascii` instead (though I don't know how that looks in the Jenkins console output).

Reply:
hi,
Thanks for your reply.
```
It doesn't work on Ubuntu 18.04. While this isn't supported on Foxy, we still need to be able to run this for Eloquent and Dashing, so we need to make this conditional on the Ubuntu version.
```
In fact, I am running CI jobs on an AMD server withUbuntu 18, and the configuration comes from the master branch, so there may be some configuration differences between the official server and my server. 
````
We don't want to totally disable the progress bar, as that doesn't give us any indication that pip hung on a download. I might suggest switching this to ascii instead (though I don't know how that looks in the Jenkins console output).
````
I tried to change the stdout of the pip progress bar to ASCII through "--progress-bar ascii", and the log is still very verbose. The change looks like the block has changed to "#", and the number of lines in the progress bar has not decreased. 

The solution may be to change the printing speed, progress bar line would be printed every 0.5 second.

Reply:
> In fact, I am running CI jobs on an AMD server with Ubuntu 18.04, and the configuration comes from the master branch, so there may be some configuration differences between the official server and my server.

Well, since this is a Dockerfile, by default you'd use the distribution specified at the top.  By default, that is 20.04, but we also have a switch to 18.04.  There, the `--progress-bar` option doesn't exist, and throws an error.

> The solution may be to change the printing speed, progress bar line would be printed every 0.5 second.

Sure, if there is a way to slow down how often it prints, that works too.  I think I would still stick with the ascii output, since there aren't any "special" characters there.